### PR TITLE
Add required RBAC on integration finalizers sub-resources

### LINKED ISSUE #1485: Missing RBAC on finalizers sub-resources when OwnerReferencesPermissionEnforcement admission is enabled
**What happened**:

On clusters, like OpenShift, where the [OwnerReferencesPermissionEnforcement](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#ownerreferencespermissionenforcement) admission controller is enabled, reconciliation of workloads failed with the following error message:

```
"error":"workloads.kueue.x-k8s.io \"pytorchjob-pytorch-simple-7a162\" is forbidden: cannot set blockOwnerDeletion if an ownerReference refers to a resource you can't set finalizers on: , <nil>"
```

**What you expected to happen**:

The Kueue service account should have the required permissions to update the finalizers.

**How to reproduce it (as minimally and precisely as possible)**:

Create a workload on a cluster that has the OwnerReferencesPermissionEnforcement admission controller enabled.

### PR DESCRIPTION
**Pull request by astefanutti**
#### What type of PR is this?

/kind bug

#### Which issue(s) this PR fixes:

Fixes #1485

#### Does this PR introduce a user-facing change?

```release-note
Add Missing RBAC on integration finalizers sub-resources
```

### CODE REVIEW

**Review by tenzen-y — COMMENTED** [review]
/ok-to-test

@astefanutti Could you update helm charts as well?

File: charts/kueue/templates/rbac/role.yaml [hunk 1]
```diff
@@ -39,6 +39,13 @@ rules:
       - patch
       - update
       - watch
+  - apiGroups:
+      - ""
+    resources:
+      - pods/finalizers
+    verbs:
+      - get
+      - patch
```
Comment (tenzen-y) [inline charts/kueue/templates/rbac/role.yaml:48]: Why do we need a `patch` permission only for the pod, instead of `update`?

Comment (astefanutti) [reply charts/kueue/templates/rbac/role.yaml]: Great catch! The Pod controller does indeed perform an update operation to remove the finalizer. I've re-pushed with the correct verb.

**Comment by tenzen-y** [comment]
@astefanutti Can you perform `make verify` and then fix errors?

**Comment by astefanutti** [comment]
> @astefanutti Can you perform `make verify` and then fix errors?

@tenzen-y I fail to find any error to fix. It seems more related to the flake tests reported in #1372?

**Comment by tenzen-y** [comment]
> > @astefanutti Can you perform `make verify` and then fix errors?
> 
> @tenzen-y I fail to find any error to fix. It seems more related to the flake tests reported in #1372?

Oops, I have seen old CI logs. That's right.
/test pull-kueue-test-e2e-main-1-28

/lgtm
/approve

**Comment by tenzen-y** [comment]
Due to ordering of manifests, we could not create a cherry-pick PR:

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_kueue/1505/pull-kueue-verify-release-0-5/1737931361345343488

@astefanutti Can you manually open a cherry-pick PR?

**Comment by astefanutti** [comment]
@tenzen-y I've created #1508. I haven't seen the differences locally, but I'll adapt the PR, so it passes the verify step.

**Comment by tenzen-y** [comment]
> @tenzen-y I've created #1508. I haven't seen the differences locally, but I'll adapt the PR, so it passes the verify step.

Thanks

**Comment by alculquicondor** [comment]
/release-note-edit
```release-note
Add Missing RBAC on finalizer sub-resources for job integrations.
```

**Pull request merged**